src/components/menu/_menu.js

```diff
@@ -0,0 +1,232 @@
+ * <hljs lang="html">
+ * <md-menu md-offset="2 0">
+ *   <!-- menu-content -->
+ * </md-menu>
+ * </hljs>
+ *
+ * @usage
+ * <hljs lang="html">
+ * <md-menu>
+ *  <md-button ng-click="$mdOpenMenu()" class="md-icon-button">
+ *    <md-icon md-svg-icon="call:phone"></md-icon>
+ *  </md-button>
+ *  <md-menu-content>
+ *    <md-menu-item><md-button ng-click="doSomething()">Do Something</md-button></md-menu-item>
+ *  </md-menu-content>
+ * </md-menu>
+ * </hljs>
+ *
+ * @param {string} md-position-mode The position mode in the form of
+             `x`, `y`. Default value is `target`,`target`. Right now the `x` axis
+             also suppports `target-right`.
+ * @param {string} md-offset An offset to apply to the dropdown after positioning
+             `x`, `y`. Default value is `0`,`0`.
+ *
+ */
+
+function MenuDirective($mdMenu) {
+  return {
+    restrict: 'E',
+    require: 'mdMenu',
+    controller: function() { }, // empty function to be built by link
+    scope: true,
+    compile: compile
+  };
+
+  function compile(tEl) {
+    tEl.addClass('md-menu');
+    tEl.children().eq(0).attr('aria-haspopup', 'true');
+    return link;
+  }
+
+  function link(scope, el, attrs, mdMenuCtrl) {
+    // Se up mdMenuCtrl to keep our code squeaky clean
+    buildCtrl();
+
+    // Expose a open function to the child scope for their html to use
+    scope.$mdOpenMenu = function() {
+      mdMenuCtrl.open();
+    };
+
+    if (el.children().length != 2) {
+      throw new Error('Invalid HTML for md-menu. Expected two children elements.');
+    }
+
+    // Move everything into a md-menu-container
+    var menuContainer = angular.element('<div class="md-open-menu-container md-whiteframe-z2"></div>');
+    var menuContents = el.children()[1];
+    menuContainer.append(menuContents);
+
+    var enabled;
+    mdMenuCtrl.enable();
+
+    function buildCtrl() {
+      mdMenuCtrl.enable = function enableMenu() {
+        if (!enabled) {
+          //el.on('keydown', handleKeypress);
+          enabled = true;
+        }
+      };
+
+      mdMenuCtrl.disable = function disableMenu() {
+        if (enabled) {
+          //el.off('keydown', handleKeypress);
+          enabled = false;
+        }
+      };
+
+      mdMenuCtrl.open = function openMenu() {
+        el.attr('aria-expanded', 'true');
+        $mdMenu.show({
+          mdMenuCtrl: mdMenuCtrl,
+          element: menuContainer,
+          target: el[0]
+        });
+      };
+
+      mdMenuCtrl.close = function closeMenu(skipFocus) {
+        el.attr('aria-expanded', 'false');
+        $mdMenu.hide();
+        if (!skipFocus) el.children()[0].focus();
+      };
+
+      mdMenuCtrl.positionMode = function() {
+        var attachment = (attrs.mdPositionMode || 'target').split(' ');
+
+        if (attachment.length == 1) { attachment.push(attachment[0]); }
+
+        return {
+          left: attachment[0],
+          top: attachment[1]
+        };
+
+      };
+
+      mdMenuCtrl.offsets = function() {
```

Comment:
I'd like all of these controller functions to have descriptions, but this one in particular. A method name of `offsets` does not immediately convey what it does.